src/java.base/share/classes/java/text/NumberFormat.java

```diff
@@ -394,6 +394,18 @@ public abstract StringBuffer format(double number,
                                         StringBuffer toAppendTo,
                                         FieldPosition pos);
 
+
+    /**
+     * Same as format(double, StringBuffer, FieldPosition), but using StringBuilder
+     * instead of StringBuffer.
+     */
+    StringBuilder format(double number,
+                         StringBuilder toAppendTo,
+                         FieldPosition pos) {
+        throw new UnsupportedOperationException("Subclasses should override this " +
```

Comment:
NumberFormat supports user-created classes, as it is abstract and has a protected constructor. This will cause exceptions in formatting users' custom NumberFormat subclasses.